src/chttpd/src/chttpd_node.erl

```diff
@@ -405,18 +405,29 @@ get_distribution_stats() ->
 message_queues() ->
     MessageQueuesAgg = [
         {couch_server, couch_server:aggregate_queue_len()},
-        {index_server, couch_index_server:aggregate_queue_len()}
+        {index_server, couch_index_server:aggregate_queue_len()},
+        {rexi_server, rexi:aggregate_server_queue_len()},
+        {rexi_buffer, rexi:aggregate_buffer_queue_len()}
     ],
-    MessageQueuesReg = lists:map(
-        fun(Name) ->
-            Type = message_queue_len,
-            {Type, Length} = process_info(whereis(Name), Type),
-            {Name, Length}
-        end,
-        registered()
-    ),
+    MessageQueuesReg = lists:filtermap(fun message_queue/1, registered()),
     MessageQueuesAgg ++ MessageQueuesReg.
 
+message_queue(rexi_server) ->
+    % Compatibility clause. Remove in 3.4+ version when singleton
+    % rexi_server is removed.
+    false;
+message_queue(Name) ->
+    Pid = whereis(Name),
+    case is_pid(Pid) of
+        true ->
+            case process_info(Pid, message_queue_len) of
+                {message_queue_len, Length} -> {true, {Name, Length}};
+                _ -> false
```

Comment:
we'd only expect `undefined` here, shouldn't we say so?

Reply:
similar for is_pid which returns either pid() or `undefined`

Reply:
agree, let's go with tighter matching, I was being too defensive but the response types are well specified here as `undefined`